src/trio/_core/_run.py

```diff
@@ -328,29 +328,29 @@ class CancelStatus:
     # effectively cancelled due to the cancel scope two levels out
     # becoming cancelled, but then the cancel scope one level out
     # becomes shielded so we're not effectively cancelled anymore.
-    effectively_cancelled: bool = attr.ib(default=False)
+    effectively_cancelled: bool = attrs.field(default=False)
 
     # The CancelStatus whose cancellations can propagate to us; we
     # become effectively cancelled when they do, unless scope.shield
     # is True.  May be None (for the outermost CancelStatus in a call
     # to trio.run(), briefly during TaskStatus.started(), or during
     # recovery from mis-nesting of cancel scopes).
-    _parent: CancelStatus | None = attr.ib(default=None, repr=False)
+    _parent: CancelStatus | None = attrs.field(default=None, repr=False, alias="parent")
```

Comment:
Why are we exposing this?

Reply:
As explained in the PR description, my main goal for this PR was to make Pyright stop complaining about constructor kwargs for these classes. These aliases + `attrs>=21.2.0` is what actually accomplishes that (though I was perfectly happy to handle some other `attrs`-related updates that were agreed upon in chat and comments).